src/views/virtualmachines/actions/VirtualMachineActionFactory.tsx

```diff
@@ -76,23 +80,23 @@ export const VirtualMachineActionFactory = {
       label: t('Cancel compute migration'),
     };
   },
-  cancelStorageMigration: (
-    vm: V1VirtualMachine,
-    vmim: V1VirtualMachineInstanceMigration,
-    isSingleNodeCluster: boolean,
-  ): Action => {
+  cancelStorageMigration: (currentStorageMigration: MigMigration): Action => {
     return {
       accessReview: {
-        group: VirtualMachineModel.apiGroup,
-        namespace: vm?.metadata?.namespace,
-        resource: VirtualMachineModel.plural,
+        group: MigMigrationModel.apiGroup,
+        namespace: DEFAULT_MIGRATION_NAMESPACE,
+        resource: MigMigrationModel.plural,
         verb: 'patch',
       },
-      cta: () => {
-        cancelMigration(vmim);
-        return rollbackStorageMigration(vm);
-      },
-      disabled: isSingleNodeCluster,
+      cta: () =>
+        k8sPatch({
+          data: [
+            { op: 'replace', path: '/spec/rollback', value: true },
+            { op: 'remove', path: '/spec/migrateState' },
```

Comment:
removing migration state seems unusual - why it's up to the UI to clean it up?

Reply:
the migplan do not support rollback and migrateState specs together. only one of the two